src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts

```diff
@@ -589,11 +597,31 @@ export class ProgrammingExerciseConfigureGradingComponent implements OnInit, OnD
     /**
      * Executes filtering on all available test cases with the specified params.
      */
-    updateTestCaseFilter = () => {
+    updateTestCaseFilter() {
         this.filteredTestCasesForTable = !this.showInactiveValue && this.testCases ? this.testCases.filter(({ active }) => active) : this.testCases;
         this.filteredTestCasesForCharts = this.filteredTestCasesForTable;
         this.backupTestCases = this.filteredTestCasesForTable;
-    };
+    }
+
+    /**
+     * Calculates the rounded points awarded for passing each test
+     */
+    updateTestPoints() {
+        this.testCasePoints = {};
+
+        if (!this.testCases) {
+            return;
+        }
+
+        const totalWeight = this.testCases.reduce((sum, testCase) => sum + testCase.weight!, 0);
```

Comment:
I suggest replacing this with a global totalWeight value as in the charts.

Then in the `updateEditedField` recalculate the total weight value only if a weight field was edited. This way you only recalculate when necessary. Technically, if you have access to the value from before (not sure) you could even update the value without iterating over the whole array but I'm not sure if you do.